library/core/src/array/mod.rs

```diff
@@ -364,3 +364,54 @@ macro_rules! array_impl_default {
 }
 
 array_impl_default! {32, T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T}
+
+#[cfg(not(bootstrap))]
+#[lang = "array"]
+impl<T, const N: usize> [T; N] {
+    /// Returns an array of the same size as self, with `f` applied to each element.
+    ///
+    /// # Examples
+    /// ```
+    /// # #![feature(array_map)]
+    /// let x = [1, 2, 3];
+    /// let y = x.map(|v| v + 1);
+    /// assert_eq!(y, [2, 3, 4]);
+    /// ```
+    #[unstable(feature = "array_map", issue = "75027")]
+    pub fn map<F, U>(self, mut f: F) -> [U; N]
+    where
+        F: FnMut(T) -> U,
+    {
+        use crate::mem::MaybeUninit;
+        struct Guard<T, const N: usize> {
```

Comment:
This type is functionally equivalent to the one used in array's `into_iter()`, just adding things to the array instead of removing things from it.  Should they be unified?

Reply:
ah sorry which in specific are you referring to? Is this one of the PR's open for implementing `IntoIterator for [T; N]`?

Reply:
There's no trait implementation (sadly), but the iterator type itself does exist: https://doc.rust-lang.org/std/array/struct.IntoIter.html

Reply:
While they are similar, this one does only iterate in one direction so unifying it might take some work. Would you think it's important to do so?

Reply:
I put it as a question since I'm not sure.  My default answer is to reduce the amount of nuanced unsafe code, but if it's awkward to merge then you can probably just leave it -- unless a real libs-impl member says otherwise, of course.

Reply:
Mm I do feel the use case is sufficiently different from the iterator, so I'm going to leave it for now then, unless there's a reason otherwise.

Reply:
I think merging that code is too much hassle to be worth it. The iterator also uses a range (to be a doubled ended iterator) which is not required here.